load_issues.py

```diff
@@ -0,0 +1,178 @@
+from pathlib import Path
+from typing import Any, Dict, List
+import os
+import textwrap
+
+import typer
+import yaml
+
+from github import Github
+from github.Issue import Issue
+from github.Repository import Repository
+
+from loguru import logger
+
+DEFAULT_CONFIG = {"owner": "uw-ssec", "repo": "codeuw", "repos": []}
+
+
+def load_config(config_file: str = ".codeuw-config.yml") -> Dict[str, Any]:
+    """
+    Load configuration file
+
+    Parameters
+    ----------
+    config_file : str, optional
+        The string path to configuration file,
+        by default ".codeuw-config.yml"
+
+    Returns
+    -------
+    Dict[str, Any]
+        The configuration dictionary
+    """
+    config_path = Path(config_file)
+    config = yaml.safe_load(config_path.read_text())
+
+    # Set defaults, in case they're not in the config file
+    for key, value in DEFAULT_CONFIG.items():
+        config.setdefault(key, value)
+
+    return config
+
+
+def setup_github() -> Github:
+    """
+    Setup Github API
+
+    Returns
+    -------
+    Github
+        Github API object
+    """
+    from github import Auth
+
+    # using an access token
+    # it'll look for environment variable GITHUB_TOKEN
+    auth = Auth.Token(os.environ.get("GITHUB_TOKEN"))
+
+    # Public Web Github
+    return Github(auth=auth)
+
+
+def parse_issue_template(source_repo: Repository) -> Dict[str, Any]:
+    """
+    Parse the issue template from the source repository
+    and output a dictionary of the template
+
+    Parameters
+    ----------
+    source_repo : Repository
+        The source repository to parse the issue template from
+
+    Returns
+    -------
+    dict
+        The dictionary of the issue template
+    """
+    issue_template = source_repo.get_contents(path=".github/ISSUE_TEMPLATE/task.yml")
+    task_template = yaml.safe_load(issue_template.decoded_content)
+
+    # Create the body markdown template
+    body_template = ""
+    for input in task_template["body"]:
+        input_id = input["id"]
+        section_string = textwrap.dedent(
+            f'### {input["attributes"]["label"]}\n\n' f"{{{input_id}}}\n\n"
+        )
+        body_template += section_string
+
+    # Setup the template dictionary
+    template_dict = {
+        "title": (task_template["title"] + "{project_name} - {title_text}").format,
+        "labels": task_template["labels"],
+        "body": body_template.format,
+    }
+
+    return template_dict
+
+
+def generate_code_uw_issues(
+    issues_with_label: List[Issue],
+    template_dict: Dict[str, Any],
+    repo: Dict[str, Any],
+    source_issue_titles: List[str],
```

Comment:
I found the use of `source` confusing. After reading the full PR I realize now that `source`=CodeUW. But since we are importing issues from repos and writing them to the CodeUW I was initially thinking that `source` referred the the repos we are importing **from** (e.g. NoisePy, AutoDoc, Caustics), as in "the source of the issues". I think renaming `source*` to `codeuw*` would make things much clearer. We could then call the other repos `sources`. WDYT?